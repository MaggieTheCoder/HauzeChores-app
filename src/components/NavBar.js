import React from "react";
import propTypes from "prop-types";
import "../styles/NavBar.css";

const NavBar = ({ user, logOut, code }) => {
  return (
    <nav className="navbar">
      {user && (
        <div>
          <p>Logged in as {user.email}</p>
          <button type="button" onClick={() => logOut()}>
            Log out
          </button>
          {code && <p>{code}</p>}
        </div>
      )}
    </nav>
  );
};

export default NavBar;

NavBar.propTypes = {
  user: propTypes.shape(),
  logOut: propTypes.func.isRequired,
  code: propTypes.string.isRequired,
};

NavBar.defaultProps = {
  user: null,
};
