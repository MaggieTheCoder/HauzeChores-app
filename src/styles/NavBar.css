.navbar {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.logout_button {
  font-family: "Menlo", sans-serif;
  font-size: 10px;
  margin-bottom: 10px;
  background-color: #C60982;
  flex-direction: row;
  border-radius: 10px;
  padding: 5px;
  color: white;
}  
.user {
  font-family: "Menlo", sans-serif;
  flex-direction: row
}             

