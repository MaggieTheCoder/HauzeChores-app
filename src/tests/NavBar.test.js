import React from "react";
import { render, screen } from "@testing-library/react";
import NavBar from "../components/NavBar";

describe("NavBar component", () => {
  const validProps = {
    user: { email: "test@test.com" },
    logout: () => {},
    code: "test-code",
  };

  test("renders properly", () => {
    const { asFragment } = render(
      <NavBar
        user={validProps.user}
        logout={validProps.logout}
        code={validProps.code}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("log out button renders correctly", () => {
    render(
      <NavBar
        user={validProps.user}
        logout={validProps.logout}
        code={validProps.code}
      />
    );

    const buttonElement = screen.getByText("LOG OUT");

    expect(buttonElement).toBeInTheDocument();
  });

  test("invite code renders when code prop is true value", () => {
    render(
      <NavBar
        user={validProps.user}
        logout={validProps.logout}
        code={validProps.code}
      />
    );
    const codeElement = screen.getByText("test-code");

    expect(codeElement).toBeInTheDocument();
  });

  test("user email to render when user prop is true value", () => {
    render(
      <NavBar
        user={validProps.user}
        logout={validProps.logout}
        code={validProps.code}
      />
    );
    const emailElement = screen.getByText(
      `Logged in as ${validProps.user.email}`
    );

    expect(emailElement).toBeInTheDocument();
  });
});
