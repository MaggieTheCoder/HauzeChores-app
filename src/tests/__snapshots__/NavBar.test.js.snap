// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders properly 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div>
      <header
        class="header"
      >
        <div
          class="header__banner"
        >
          banners
        </div>
        <h1
          class="header__h1"
        >
          HOUZE
        </h1>
        <h3
          class="header__h3"
        >
          Just get it done!
        </h3>
      </header>
      <p
        class="user"
      >
        Logged in as test@test.com
      </p>
      <button
        class="logout_button"
        type="button"
      >
        LOG OUT
      </button>
      <p>
        test-code
      </p>
      <footer
        class="footer"
      >
        <p
          class="footer-text"
        >
          Footer
        </p>
        <img
          alt="Houze logo"
          class="app-logo"
          src="Homepage-logo-purple.svg"
        />
      </footer>
    </div>
  </nav>
</DocumentFragment>
`;
